pyrefinebio/dataset.py

```diff
@@ -5,6 +5,43 @@
 
 
 class Dataset:
+    """Dataset
+
+    Datasets are collections of experiments and their samples.
+    The dataset can be processed and then downloaded.
+
+    Create and save a dataset
+
+        >>> import pyrefinebio
+        >>> dataset = pyrefinebio.Dataset(email_address="example@refine.bio", data={"SRP003819": ["SRR069230"]})
+        >>> dataset = dataset.save()
+
+    Get a dataset that has been saved
+
+        >>> import pyrefinebio
+        >>> id = "dataset id <guid>"
+        >>> dataset = pyrefinebio.Dataset.get(id)
+
+    Start processing a dataset
+
+        >>> import pyrefinebio
+        >>> dataset = pyrefinebio.Dataset(...).save()
+        >>> dataset.process()
+
+    Check if a dataset is finished processing
+
+        >>> import pyrefinebio
+        >>> dataset = pyrefinebio.Dataset(...).save()
+        >>> dataset.process()
+        >>> dataset.check()
+
+    Download a processed dataset
+
+        >>> import pyrefinebio
+        >>> dataset = pyrefinebio.Dataset(...).save()
+        >>> dataset.process()
+        >>> dataset.download("~/datasets/my_dataset.zip")
+    """
```

Comment:
`process()` calls `save()`, so I think most of your examples don't need it. I think providing an API token should be included in these examples since they won't work without it. But now I'm wondering how the token gets provided at all. Do we need an issue for that?